File: src_local/utils/persona_loader.py
import importlib.util
import logging
from pathlib import Path

logger = logging.getLogger(__name__)

class PersonaLoader:
    """
    🏛️ Mobilizador de Personas PhD.
    Especialista em carregamento dinâmico de identidades soberanas do disco,
    garantindo que cada Agente PhD seja instanciado com o contexto correto.
    """
    
    @staticmethod
    def mobilize_all(target_project_root, orchestrator):
        """
        🚀 Varre os diretórios de agentes e mobiliza a junta PhD correspondente.
        Realiza a injeção atômica do Maestro em cada identidade carregada.
        """
        engine_root = Path(__file__).parent.parent
        agents_base = engine_root / "agents"
        
        count = 0
        # Varre subdiretórios de stacks (Python, Flutter, Kotlin)
        for stack_dir in agents_base.iterdir():
            if not stack_dir.is_dir() or stack_dir.name in ["__pycache__", "Support"]:
                continue
                
            # rglob para busca recursiva em todos os subdiretórios técnicos
            for f in stack_dir.rglob("*.py"):
                if f.name == "__init__.py" or "__pycache__" in str(f): continue
                try:
                    # Gera um nome de módulo único baseado no caminho relativo para evitar conflitos
                    relative_path = f.relative_to(agents_base)
                    module_name = f"{stack_dir.name}_" + "_".join(relative_path.parts[1:]).replace(".py", "")
                    
                    spec = importlib.util.spec_from_file_location(module_name, f)
                    module = importlib.util.module_from_spec(spec)
                    spec.loader.exec_module(module)
                    for attr in dir(module):
                        if attr.endswith("Persona") and attr != "BaseActivePersona":
                            persona_cls = getattr(module, attr)
                            orchestrator.add_persona(persona_cls(target_project_root))
                            count += 1
                except Exception as e:
                    logger.error(f"❌ Falha ao mobilizar {f.name} ({stack_dir.name}): {e}")
        return count

File: src_local/utils/persona_loader.ts
import winston from "winston";

const logger = winston.child({ module: "PersonaLoader" });

export class PersonaLoader {
    static async mobilizeAll(projectRoot: string, orchestrator: any) {
        logger.info("🎭 [Loader] Iniciando mobilização em massa de PhDs (Bun)...");
        const loader = new PersonaLoader();
        await loader.loadPersonas(orchestrator);
    }

    async loadPersonas(orchestrator: any) {
        try {
            const manifestPath = path.join(__dirname, "persona_manifest.json");
            const manifest = JSON.parse(fs.readFileSync(manifestPath, 'utf-8'));

            logger.info(`🎭 [Loader] Carregando ${manifest.personas.length} PhDs do manifesto...`);

            const { DynamicPersona } = await import("../agents/dynamic_persona.ts");
            const { TestifyPersona } = await import("../agents/Support/testify_persona.ts");

            for (const metadata of manifest.personas) {
                let persona;
                if (metadata.name === "Testify") {
                    persona = new TestifyPersona(orchestrator.projectRoot.toString());
                } else {
                    persona = new DynamicPersona(metadata, orchestrator.projectRoot.toString());
                }
                orchestrator.addPersona(persona);
            }

            logger.info("🎭 [Loader] Mobilização em massa concluída.");
        } catch (e) {
            logger.error(`❌ [Loader] Falha ao carregar manifesto de personas: ${e}`);

            // Fallback for critical ones if manifest fails
            orchestrator.addPersona({ name: "Director", role: "Strategic Lead" });
        }
    }
}

import * as fs from 'fs';
import * as path from 'path';
